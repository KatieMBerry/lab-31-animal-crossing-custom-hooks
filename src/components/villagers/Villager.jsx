import React from 'react';
import PropTypes from 'prop-types';
// import styles from './Villager.css';

const Villager = ({ image, name }) => (
    <figure data-testid="villager"
        className={StyleSheet.Villager}>
        <img src={image} alt={name} />
        <figcaption>{name}</figcaption>
    </figure>
)

Villager.propTypes = {
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}

export default Villager;
