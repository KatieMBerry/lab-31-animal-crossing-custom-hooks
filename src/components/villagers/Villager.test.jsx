import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Villager from './Villager';

describe('Villager component', () => {
    afterEach(() => cleanup());
    it('renders Villager', () => {
        const { asFragment } = render(<Villager
            name='Agnes'
            image='https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432'
        />);
        expect(asFragment()).toMatchSnapshot();
    });
});
