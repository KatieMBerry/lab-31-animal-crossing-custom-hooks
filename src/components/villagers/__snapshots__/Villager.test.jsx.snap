// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager component renders Villager 1`] = `
<DocumentFragment>
  <figure
    data-testid="villager"
  >
    <img
      alt="Agnes"
      src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
    />
    <figcaption>
      Agnes
    </figcaption>
  </figure>
</DocumentFragment>
`;
